src/diskImage.cpp

```diff
@@ -12,14 +17,63 @@ void DiskImage::setPath(const std::filesystem::path& path)
     if (path.extension() != ".iso")
         throw std::runtime_error("Disk Image must have ISO extension");
 
+    // Verify checksum only if the image is known.
+    if (isKnownImage(path)) {
+        if (!hasVerifiedChecksum(path))
+            throw std::runtime_error("Disk Image checksum isn't valid");
+    }
+
     m_path = path;
 }
 
-bool DiskImage::isKnownImage()
+bool DiskImage::isKnownImage(const std::filesystem::path& path)
 {
     for (const auto& image : m_knownImageFilename)
-        if (m_path.filename() == image)
+        if (path.filename().string() == image) {
+            LOG_TRACE("Disk image is recognized");
             return true;
+        }
+
+    LOG_TRACE("Disk image is unknown. Maybe you're using a custom image or "
+              "changed the default name?");
+    return false;
+}
+
+bool DiskImage::hasVerifiedChecksum(const std::filesystem::path& path)
+{
+    using namespace CryptoPP;
+    using namespace std;
+
+    LOG_TRACE("Verifying disk image checksum... This may take a while");
+
+    map<string, string> hash_map = {
```

Comment:
A `map` type uses a b-type tree to store the data. Wouldn't a hashmap be an better option? It can be implemented with `unordered_map`.